### PR TITLE
[FEAT] Add the parsing specify addresses function

### PR DESCRIPTION
**Pull request by oneleo**
### New feature
Add the use of eth + dot + domain or btc + dot + domain
parsing specify the eth or btc address of the token.eth domain.

### How to verify?
Please read the new [Readme.md](https://github.com/NIC619/offchain-resolver/blob/IRARA/resolve_only_btc_or_eth_addresses/README.md) file.

Thanks!

### CODE REVIEW

File: packages/client/src/index.ts [hunk 1]
```diff
@@ -21,9 +21,42 @@ const provider = new ethers.providers.JsonRpcProvider(options.provider, {
...
+    // E.g.: btc.token.eth -> real domain = token.eth
+    // But: foo.token.eth -> real domain still = foo.token.eth
+    name = domainArray.slice(1, domainArray.length).join('.');
+  }
+
+  console.log(`Resolving ${name} domain...`);
+  const resolver = await provider.getResolver(name);
+
+  if (!resolver) {
+    console.log(`No resolver contract or gatway server found`);
```
Comment (NIC619) [inline packages/client/src/index.ts]:
```suggestion
    console.log("No resolver contract or gateway server found");
```

File: packages/client/src/index.ts [hunk 2]
```diff
@@ -21,34 +21,119 @@ const provider = new ethers.providers.JsonRpcProvider(options.provider, {
...
-    console.log(`Content: ${content}`);
+  let name = program.args[0];
+
+  // Try to get the coin name from first element of domain name to resolve
+  if (isGivenCoin(name)) {
+    // Coin resolution is successful, print the specified coin address
+    resolveGivenCoin(name);
   } else {
-    console.log('no resolver found');
+    // Coin resolution is fails, print the all info
```
Comment (NIC619) [inline packages/client/src/index.ts]:
```suggestion
    // Coin resolution fails, print all info
```

File: packages/client/src/index.ts [hunk 3]
```diff
@@ -21,34 +21,119 @@ const provider = new ethers.providers.JsonRpcProvider(options.provider, {
...
 })();
 
+// Decode address to EIP-2304 onchain hex string
 function addressToOnchainHex(address: string, coinType: string): string {
-  // Decode to EIP-2304 onchain hex string
   const onchain = formatsByName[coinType].decoder(address);
   return `0x${onchain.toString('hex')}`;
 }
+
+// Judge the domain contains coin name
```
Comment (NIC619) [inline packages/client/src/index.ts]:
```suggestion
// Determine if the first name in the domain is a name of a coin
```

File: packages/client/src/index.ts [hunk 4]
```diff
@@ -21,34 +21,119 @@ const provider = new ethers.providers.JsonRpcProvider(options.provider, {
...
+function isGivenCoin(domain: string): boolean {
+  // Try to get the coin name from first element of domain name to resolve
```
Comment (NIC619) [inline packages/client/src/index.ts]:
```suggestion
  // Try to get the first name in the domain
```

File: packages/client/src/index.ts [hunk 5]
```diff
@@ -21,34 +21,119 @@ const provider = new ethers.providers.JsonRpcProvider(options.provider, {
...
+  const domainArray = domain.split('.');
+  if (domainArray.length <= 1) {
+    console.log(`[Error] Domain must have at least one dot "."`);
+    process.exit(0);
+  }
+  const coinName = domainArray[0]; // Get first element
+  return isCoin(coinName);
+}
+
+// Judge the string is coin name
```
Comment (NIC619) [inline packages/client/src/index.ts]:
```suggestion
// Determine if the name is name of a coin
```

File: packages/client/src/index.ts [hunk 6]
```diff
@@ -21,34 +21,119 @@ const provider = new ethers.providers.JsonRpcProvider(options.provider, {
...
+function isCoin(name: string): boolean {
+  const coinType = getCoinType(name);
+  return coinType === null ? false : true;
+}
+
+// Transfer coin name to coin type number
```
Comment (NIC619) [inline packages/client/src/index.ts]:
```suggestion
// Get coin type from the name
```

File: packages/client/src/index.ts [hunk 7]
```diff
@@ -21,34 +21,119 @@ const provider = new ethers.providers.JsonRpcProvider(options.provider, {
...
+// Resolve the specific coin address from domain name
+async function resolveGivenCoin(domain: string) {
+  const domainArray = domain.split('.');
+  // Get first element as coin name
+  const coinName = domainArray[0];
+  const coinType = getCoinType(coinName)!;
+  const coinNameToUpper = coinName.toUpperCase();
+
+  // Change name to the real domain.
+  // E.g.: btc.token.eth -> real domain = token.eth
```
Comment (NIC619) [inline packages/client/src/index.ts]:
```suggestion
  // Remove coin name from domain
  // E.g.: btc.token.eth -> token.eth
```

File: packages/client/src/index.ts [hunk 8]
```diff
@@ -21,34 +21,119 @@ const provider = new ethers.providers.JsonRpcProvider(options.provider, {
...
+  const name = domainArray.slice(1, domainArray.length).join('.');
+
+  console.log(`Resolving ${name} domain...`);
+  const resolver = await provider.getResolver(name);
+
+  if (!resolver) {
+    console.log(`[Error] No resolver contract or gateway server found`);
```
Comment (NIC619) [inline packages/client/src/index.ts]:
```suggestion
    console.log(`[Error] No resolver contract found`);
```

File: packages/client/src/index.ts [hunk 9]
```diff
@@ -21,34 +21,119 @@ const provider = new ethers.providers.JsonRpcProvider(options.provider, {
...
+  );
+}
+
+// Resolve all info from domain name
+async function resolveAllData(domain: string) {
+  console.log(`Resolving ${domain} domain...`);
+  const resolver = await provider.getResolver(domain);
+
+  if (!resolver) {
+    console.log(`[Error] No resolver contract or gateway server found`);
```
Comment (NIC619) [inline packages/client/src/index.ts]:
```suggestion
    console.log(`[Error] No resolver contract found`);
```

**Review by NIC619 — APPROVED** [review]
Just some nits

File: packages/client/src/index.ts (hunk 8 above)
Comment (oneleo) [reply packages/client/src/index.ts]: Yes! You're right.
Error occurring when resolve if gateway is down,
an empty resolver value will not be returned.

**Pull request merged**